sdk/storage/storage-queue/RecordAndPlayback.md

```diff
@@ -0,0 +1,192 @@
+# GUIDELINES FOR RECORD AND PLAYBACK
+
+## NOCK [for node tests]
+
+- [nock](https://www.npmjs.com/package/nock)-package is being used to test modules that perform HTTP requests.
+- To mock an existing live system, we record and playback the HTTP calls using `nock.recorder`.
+- Recording relies on intercepting real requests and responses and then persisting them for later use.
+
+---
+
+## NISE [for browser tests]
+
+- Nock has no support for browsers. For them, we're using [nise](https://www.npmjs.com/package/nise).
+- Nise works in a way similar to Nock, intercepting HTTP requests and mocking responses.
+- Unlike Nock, Nise does not have a native record/playback feature.
+- Some Nise functions are being overwritten to enable record and playback.
```

Comment:
Which functions get overrided? Will this still work for future versions of Nise?

Reply:
@kinelski

Reply:
We are overriding a few methods from FakeXMLHttpRequest in Nise:
- In record mode: open, send and onreadystatechange.
- In playback mode: send.

The reasons behind it are well explained in comments in `recorder.ts`.

It should work for future versions. These methods are guaranteed to exist because they must follow an [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest) structure.

Also, our changes are not too invasive. We are simply including code into Nise functions, but we are making sure their original implementations are being called. So, even if their code changes in future versions, the changes will executed as expected.